gluecodium/src/main/resources/templates/jni/utils/JniExceptionThrowerHeader.mustache

```diff
@@ -0,0 +1,59 @@
+{{!!
+  !
+  ! Copyright (C) 2016-2020 HERE Europe B.V.
+  !
+  ! Licensed under the Apache License, Version 2.0 (the "License");
+  ! you may not use this file except in compliance with the License.
+  ! You may obtain a copy of the License at
+  !
+  !     http://www.apache.org/licenses/LICENSE-2.0
+  !
+  ! Unless required by applicable law or agreed to in writing, software
+  ! distributed under the License is distributed on an "AS IS" BASIS,
+  ! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ! See the License for the specific language governing permissions and
+  ! limitations under the License.
+  !
+  ! SPDX-License-Identifier: Apache-2.0
+  ! License-Filename: LICENSE
+  !
+  !}}
+{{>java/CopyrightHeader}}
+
+#pragma once
+
+#include "JniReference.h"
+
+{{#internalNamespace}}
+namespace {{.}}
+{
+{{/internalNamespace}}
+namespace jni
+{
+    class JniExceptionThrower
+    {
+    public:
+        JniExceptionThrower(JNIEnv* jni_env) : m_jni_env(jni_env)
+        {
+        }
+
+        ~JniExceptionThrower()
+        {
+            if (m_exception)
+            {
+                m_jni_env->Throw(static_cast<jthrowable>(m_exception.release()));
+            }
+        }
+
+        void register_exception(JniReference<jobject> exception)
+        {
+            m_exception = std::move(exception);
+        }
+    private:
+        JNIEnv* m_jni_env;
+        JniReference<jobject> m_exception = nullptr;
```

Comment:
I don't think that initialisation is necessary, it's unique_ptr after all and should be inited automatically

Reply:
Done